src/Models/TerminusModel.php

```diff
@@ -0,0 +1,127 @@
+{
+    use RequestAwareTrait;
+
+    /**
+     * @var string
+     */
+    public $id;
+    /**
+     * @var array Arguments for fetching this model's information
+     */
+    protected $args = [];
+    /**
+     * @var object
+     */
+    protected $attributes;
+
+    /**
+     * @var string The URL at which to fetch this model's information
+     */
+    protected $url;
+
+    /**
+     * Object constructor
+     *
+     * @param object $attributes Attributes of this model
+     * @param array $options Options with which to configure this model
+     */
+    public function __construct($attributes = null, array $options = [])
+    {
+        if (is_object($attributes)) {
+            $this->attributes = $this->parseAttributes($attributes);
+            if (isset($this->attributes->id)) {
+                $this->id = $this->attributes->id;
+            }
+        } else {
+            $this->attributes = (object)[];
+        }
+    }
+
+    /**
+     * Fetches this object from Pantheon
+     *
+     * @param array $args Params to pass to request
+     * @return TerminusModel $this
+     */
+    public function fetch(array $args = [])
+    {
+        $options = array_merge(
+            ['options' => ['method' => 'get',],],
+            $this->args,
+            $args
+        );
+        $results = $this->request->request($this->getUrl(), $options);
+        $this->attributes = (object)array_merge(
+            (array)$this->attributes,
+            (array)$this->parseAttributes($results['data'])
+        );
+
+        return $this;
+    }
+
+    /**
+     * Retrieves attribute of given name
+     *
+     * @param string $attribute Name of the key of the desired attribute
+     * @return mixed Value of the attribute, or null if not set.
+     */
+    public function get($attribute)
+    {
+        if ($this->has($attribute)) {
+            return $this->attributes->$attribute;
+        }
+        return null;
+    }
+
+    /**
+     * Checks whether the model has an attribute
+     *
+     * @param string $attribute Name of the attribute key
+     * @return boolean True if attribute exists, false otherwise
+     */
+    public function has($attribute)
+    {
+        $isset = isset($this->attributes->$attribute);
+        return $isset;
+    }
+
+    /**
+     * Sets an attribute
+     *
+     * @param string $attribute Name of the attribute key
+     * @param mixed $value The value to assign to the attribute
+     */
+    public function set($attribute, $value)
+    {
+        $this->attributes->$attribute = $value;
+    }
+
+    /**
+     * Modify response data between fetch and assignment
+     *
+     * @param object $data attributes received from API response
+     * @return object $data
+     */
+    protected function parseAttributes($data)
+    {
+        return $data;
+    }
+
+    /**
+     * Get the URL for this model
+     *
+     * @return string
+     */
+    protected function getUrl()
```

Comment:
Great question! It's because the `Workflows` collection was hard as hell to test otherwise. I think there's no harm in making both public. It's not strictly necessary within the code but it's not a totally unreasonable thing to expose.